src/service/filtering/filtering_processing.py

```diff
@@ -0,0 +1,168 @@
+            raise errors.IllegalParameterError(f"No such filter field: {field}")
+        column = columns[field]
+        minlen = analyzers.get_minimum_query_length(column.filter_strategy)
+        if minlen and len(querystring) < minlen:
+            raise errors.IllegalParameterError(
+                f"Filter field '{field}' requires a minimum query length of {minlen}")
+
+        # Translates the query field name to an available column name if necessary.
+        # For example, translates query field name '1' to a valid column name 'col_1_val'.
+        field = trans_field_func(field) if trans_field_func else field
+
+        fs.append(
+            field,
+            column.type,
+            querystring,
+            analyzers.get_analyzer(column.filter_strategy),
+            column.filter_strategy
+        )
+    return fs
+
+
+async def get_filters(
+        r: Request,
+        arango_coll: str,
+        coll_id: str,
+        load_ver: str,
+        load_ver_override: bool,
+        data_product: str,
+        columns: list[col_models.AttributesColumn],
+        view_name: str = None,
+        count: bool = False,
+        sort_on: str = None,
+        sort_desc: bool = False,
+        filter_conjunction: bool = True,
+        match_spec: SubsetSpecification = None,
+        selection_spec: SubsetSpecification = None,
+        keep: dict[str, set[col_models.ColumnType]] = None,
+        keep_filter_nulls: bool = False,
+        skip: int = 0,
+        limit: int = 1000,
+        start_after: str = None,
+        trans_field_func: Callable[[str], str] = None,
+) -> FilterSet:
+    """
+    Constructs a FilterSet and applies filters using the provided filter query and columns.
+
+    Requires a 'get_columns_func' to generate columns, represented as a dictionary mapping column name/ID to
+    columnar_attribs_common_models.AttributesColumn objects.
+
+    r - The request.
+    arango_coll - The name of the Arango collection.
+    coll_id - The collection ID.
+    load_ver - The load version.
+    load_ver_override - Whether or not to override the load version.
+    data_product - The data product id.
+    columns - list of columnar_attribs_common_models.AttributesColumn objects to be filtered on.
+    view_name - The name of the view to use for filtering.
+    count - Whether or not to return the count of matching documents.
+    sort_on - The name of the field to sort on.
+    sort_desc - Whether or not to sort in descending order.
+    filter_conjunction - Whether or not to use a conjunction for filtering.
+    match_spec - A subset specification for matching.
+    selection_spec - A subset specification for selection.
+    keep - A dictionary mapping column names to sets of column types to keep.
+    keep_filter_nulls - Whether or not to keep null values when filtering.
+    skip - The number of documents to skip.
+    limit - The maximum number of documents to return.
+    start_after - The value of the field to start after.
```

Comment:
```suggestion

    skip - the number of records to skip. Use this parameter wisely, as paging
        through records via increasing skip incrementally is an O(n^2) operation.
    limit - the maximum number of records to return. 0 indicates no limit, which is usually
        a bad idea.
    start_after - skip any records prior to and including this value in the `sort_on` field,
        which should contain unique values.
        It is strongly recommended to set up an index that the query can use to skip to
        the correct starting record without a table scan. This parameter allows for
        non-O(n^2) paging of data.
        start_after is not currently implemented for the case where any filters are appended.
```

That last sentence is going to need to be fixed for heatmaps to work properly, I forgot about that